lib/ui/urlbar.js

```diff
@@ -7,25 +7,60 @@ const EXPORTED_SYMBOLS = ['Urlbar'];
 function Urlbar(opts) {
   this.win = opts.win;
   this.events = opts.events;
-
-  this.onKeyDown = this.onKeyDown.bind(this);
+  this.privateBrowsingUtils = opts.privateBrowsingUtils;
 }
 
 Urlbar.prototype = {
   init: function() {
     this.el = this.win.document.getElementById('urlbar');
-
-    this.el.addEventListener('keydown', this.onKeyDown);
   },
   destroy: function() {
     delete this.el;
     delete this.win;
   },
-  onKeyDown: function(evt) {
-    // check the key event:
-    // if it's a printable key, fire 'printable-key'
-    // if it's a navigational key, fire 'navigational-key'
-    // if we don't know what to do, return true or false
+  // Return `true` to prevent the existing Gecko key handler code from running.
+  onKeyPress: function(evt) {
+    // We want to send printable keys to the recommendation server, but
+    // we also want the existing popup code to search history and fetch
+    // suggestions with that keystroke.
+    if (this.isPrintableKey(evt)) {
+      this.handlePrintableKey(evt);
+    }
+    // In #43 we might want to intercept navigational keys and prevent the
+    // existing code from handling them, depending on whether the
+    // recommendation is shown or not.
+  },
+  isPrintableKey: function(evt) {
+    // Criteria applied to decide if a key is printable:
+    //
+    // 1. Based on mdn.io/KeyboardEvent/key#Key_values, any printable key
+    //    will have an event.key value of length 1, that is, just the char
+    //    itself. Other keys will have longer string values, like 'Escape'.
+    //
+    // 2. If modifiers other than Shift are pressed, the user might be typing
+    //    a shortcut key combination. So, if those are pressed, do nothing.
+    //
+    // 3. If we're in private browsing mode, respect that and do nothing.
```

Comment:
Not sure this is the correct place for this; seems like a separate concern. Maybe as an `if` block wrapping `handlePrintableKey`?